homeassistant/components/remootio/cover.py

```diff
@@ -0,0 +1,184 @@
+"""Support for by a Remootio device controlled garage door or gate."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aioremootio import Event, EventType, Listener, RemootioClient, State, StateChange
+
+from homeassistant.components import cover
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ENTITY_ID, ATTR_NAME, CONF_DEVICE_CLASS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import ATTR_SERIAL_NUMBER, CONF_SERIAL_NUMBER, DOMAIN, REMOOTIO_CLIENT
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up an ``RemootioCover`` entity based on the given configuration entry."""
+
+    _LOGGER.debug("config_entry [%s]", config_entry.as_dict())
+    _LOGGER.debug(
+        "hass.data[%s][%s] [%s]]",
+        DOMAIN,
+        config_entry.entry_id,
+        hass.data[DOMAIN][config_entry.entry_id],
+    )
+
+    serial_number: str = config_entry.data[CONF_SERIAL_NUMBER]
+    device_class: str = config_entry.data[CONF_DEVICE_CLASS]
+    remootio_client: RemootioClient = hass.data[DOMAIN][config_entry.entry_id][
+        REMOOTIO_CLIENT
+    ]
+
+    async_add_entities(
+        [
+            RemootioCover(
+                serial_number, config_entry.title, device_class, remootio_client
+            )
+        ]
+    )
+
+
+class RemootioCover(cover.CoverEntity):
+    """Cover entity which represents an Remootio device controlled garage door or gate."""
+
+    _remootio_client: RemootioClient
+
+    def __init__(
+        self,
+        unique_id: str,
+        name: str,
+        device_class: str,
```

Comment:
Device class is now enums: https://github.com/home-assistant/core/blob/689b347904857c77c1346f3f1e971f2dc31d49ca/homeassistant/components/cover/__init__.py#L51